src/components/typeInput/TypeInput.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { isWordComplete } from "./utils";
+
+interface TypeInputProps {
+    sentenceWords: string[], 
+    currentTargetWordIndex: number, 
+    setDone: React.Dispatch<React.SetStateAction<string[]>>, 
+    setCurrentWordDoneCharacters: React.Dispatch<React.SetStateAction<string>>, 
+    setCurrentTargetWordIndex: React.Dispatch<React.SetStateAction<number>>

```

Comment:
you are cupling your code to the type of ` React.Dispatch`, just use regular callback syntax.
example:
```suggestion
    setDone: (done: string[]) => void,  
    setCurrentWordDoneCharacters: React.Dispatch<React.SetStateAction<string>>, //do this here too
    setCurrentTargetWordIndex: React.Dispatch<React.SetStateAction<number>> //do this here too
```